### ISSUE TITLE
Authentication failure (cqlsh broken/python)

### ISSUE THREAD

**Issue by hjacobs** [zalando/planb-cassandra#46]
I assume the `ADMIN_PASSWORD` cannot be set in `planb-cassandra.sh`, because of:

```
I have no name!@ip-172-31-128-11:/$ cqlsh -u cassandra -p cassandra 

Warning: Cannot create directory at `/.cassandra`. Command history will not be saved.

Connection error: ('Unable to connect to any servers', {'127.0.0.1': TypeError('ref() does not take keyword arguments',)})
```

**Comment by hjacobs** [comment]
Current workaround: use the "known-good" Docker image registry.opensource.zalan.do/stups/planb-cassandra:cd60

**Comment by hjacobs** [comment]
Added option to facilitate the workaround:

```bash
$ ./create_cluster.py --docker-image registry.opensource.zalan.do/stups/planb-cassandra:cd60 --cluster-name test eu-central-1

**Comment by MALPI** [comment]
It's reported in https://issues.apache.org/jira/browse/CASSANDRA-11850 and should be fixed with 2.1.16

**Comment by a1exsh** [comment]
The workaround doesn't work unfortunately, I'm getting this error instead:
```
$ CQLSH_NO_BUNDLED=true cqlsh              

Python Cassandra driver not installed, or not on PYTHONPATH.
You might try "pip install cassandra-driver".

Python: /usr/bin/python
Module load path: ['/usr/bin', '/usr/lib/python2.7', '/usr/lib/python2.7/plat-x86_64-linux-gnu', '/usr/lib/python2.7/lib-tk', '/usr/lib/python2.7/lib-old', '/usr/lib/python2.7/lib-dynload', '/usr/local/lib/python2.7/dist-packages', '/usr/lib/python2.7/dist-packages']

Error: No module named cassandra
```

This issue cannot be fixed by reverting Cassandra version back, because it's caused by use of a newer Python version.  Now looking for a workable fix.

**Comment by a1exsh** [comment]
Please see #47.

**Comment by mohabusama** [comment]
We had similar issue in zmon-worker. Issue was resolved by upgrading cassandra-driver
cassandra-driver==3.6.0  https://github.com/zalando-zmon/zmon-worker/blob/master/requirements.txt#L27

And specifying `protocol_version` (3 works with plan-b cassandra) when connecting to cluster:
https://github.com/zalando-zmon/zmon-worker/blob/master/zmon_worker_monitor/builtins/plugins/cassandra_wrapper.py#L56

**Comment by a1exsh** [comment]
Thanks for the pointers, but unfortunately, with cqlsh it's a bit more tricky than just with the driver.  If I just update the driver, then cqlsh can't start due to some missing imports.  If I update cqlsh using pip as well, I get various protocol mismatch errors and I can't seem to find a way to override the protocol, w/o patching some file.

In the end I could make it work by patching the cqlsh installed with pip to change `DEFAULT_PROTOCOL_VERSION` constant from 4 to 3.  I think this is not significantly better than the hack I've already come up with. :-)

**Comment by a1exsh** [comment]
Fixed in #86
